structuredheadings/plugin.js

```diff
@@ -331,10 +335,11 @@
           }
 
           // apply the correct bulletstyle for all numbered headings
+          editor.fire("lockSnapshot");
           editor.execCommand("reapplyStyle", value);
           // set the combo box value
           this.setValue(value, value);
-
+          editor.fire("unlockSnapshot");
```

Comment:
This is a shady way to avoid further refactoring for now.  the execCommand also creates snapshots, so without locking & unlocking, we end up going from:

unnumbered -> has some of the numbering classes set up -> has all the numbering classes setup, 

the lock ends up combining the latter two snapshots for now.